### PR TITLE
feat: Basic gravity with floor collisions

### PR DESCRIPTION
**Pull request by OriolSors**
ClickUp task: https://app.clickup.com/t/865byn527

We implemented the basic physics required for the 1st milestone (raw gravity and floor collisions)

### CODE REVIEW

File: Source/DataModels/Components/ComponentMockState.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+#include "Geometry/LineSegment.h"
+#include "Geometry/Ray.h"
+#include "Physics/Physics.h"
+
+ComponentMockState::ComponentMockState(bool active, GameObject* owner)
+	: Component(ComponentType::MOCKSTATE, active, owner, true)
+{
+	isWinState = false;
+	isFailState = false;
```
Comment (gerard-queralt) [inline Source/DataModels/Components/ComponentMockState.cpp:17]: This both should be in the initializer list.

File: Source/DataModels/Components/ComponentMockState.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	void SetIsWinState(bool isWinState);
+	void SetIsFailState(bool isFailState);
+
+private:
+
+	bool isWinState;
+	bool isFailState;
+};
+
+inline bool ComponentMockState::GetIsWinState()
```
Comment (gerard-queralt) [inline Source/DataModels/Components/ComponentMockState.h:31]: All getters should be const
```suggestion
inline bool ComponentMockState::GetIsWinState() const
```
Same for the rest.

File: Source/DataModels/Components/ComponentPlayer.h [hunk 3]
```diff
@@ -12,5 +13,7 @@ class ComponentPlayer :
 
 	void SaveOptions(Json& meta) override;
 	void LoadOptions(Json& meta) override;
+
+private:
```
Comment (gerard-queralt) [inline Source/DataModels/Components/ComponentPlayer.h:17]: This is not needed

File: Source/DataModels/Components/ComponentRigidBody.cpp [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+#include <iostream>
+
+
+ComponentRigidBody::ComponentRigidBody(bool active, GameObject* owner)
+	: Component(ComponentType::RIGIDBODY, active, owner, true)
+{
+	isKinematic = true;
+	m = 1.0f;
+	g = 9.81;
+	v0 = float3(0.0f, 0.0f, 0.0f);
```
Comment (gerard-queralt) [inline Source/DataModels/Components/ComponentRigidBody.cpp:28]: All of these members should go in initializer lists.

File: Source/DataModels/Components/ComponentRigidBody.h [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+private:
+
+	bool isKinematic;
+	float m;
+	float g;
+	float3 v0;
+};
+
+inline bool ComponentRigidBody::GetIsKinematic()
```
Comment (gerard-queralt) [inline Source/DataModels/Components/ComponentRigidBody.h:30]: Getters should be const.

File: Source/DataModels/GameObject/GameObject.h [hunk 6]
```diff
@@ -240,3 +255,7 @@ inline void GameObject::setDrawBoundingBoxes(bool newDraw)
 	drawBoundingBoxes = newDraw;
 }
 
+inline bool GameObject::CompareTag(const std::string& commingTag)
```
Comment (gerard-queralt) [inline Source/DataModels/GameObject/GameObject.h:258]:
```suggestion
inline bool GameObject::CompareTag(const std::string& commingTag) const
```

**Review by gerard-queralt — CHANGES_REQUESTED** [review]

**Review by gerard-queralt — APPROVED** [review]

**Review by Dedado3000 — APPROVED** [review]
Nice implementation of gravity and lose/win

**Pull request merged**